modules/_labs/pagination/react/lib/Pagination/usePaginationModel.ts

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+
+import {buildPageRange} from './buildPageRange';
+import {PaginationModel} from './types';
+
+export type UsePaginationModelConfig = {
+  lastPage: number;
+  firstPage?: number;
+  initialCurrentPage?: number;
+  onPageChange?: (pageNumber: number) => void;
+  rangeSize?: number;
+};
+
+export const usePaginationModel = ({
+  firstPage = 1, // Q: Should this be a prop, or leave it out?
```

Comment:
`firstPage` isn't really state, but it is part of the model. The model is what's shared between components, so it has to exist here. If not, you'll be violating the concept of a model in a compound component and you'll need _another_ context to share this between subcomponents.